### PR TITLE
[WIP] Add agentendpoint

### PR DESCRIPTION
**Pull request by adjackura**
Sync in v1alpha1 agentendpoint
branch ospatch to start using agentendpoint, move old ospatch to ospatchog and leave it mainly untouched (github didn't understand this completly it seems)
ospatch work is still not complete, everything compiles now but writing state to disk and recovering from reboot are still not implemented (I will want to implement these for tasks more globally)

Please review the main logic in agentendpoint.go

### CODE REVIEW

File: agentendpoint/agentendpoint.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+			return
+		}
+
+		task := startRes.GetTask()
+		if task == nil {
+			return
+		}
+
+		switch task.GetTaskType() {
+		case agentendpointpb.TaskType_APPLY_PATCHES:
```
Comment (wj-chen) [inline agentendpoint/agentendpoint.go]: We don't need a return statement somewhere inside here to exit runTask?

File: agentendpoint/agentendpoint.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+)
+
+// Client is a an agentendpoint client.
+type Client struct {
+	raw *agentendpoint.Client
+}
+
+var client *Client
+
+// GetClient returns the already existing client or creates a new one.
```
[the remark this answers is not in the document]
Comment (adjackura) [reply agentendpoint/agentendpoint.go]: Well I was missing the whole assignment bit, but no this is not threadsafe but it is not meant to be run in goroutines.

File: agentendpoint/patch_linux.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package agentendpoint
+
+import (
```
[the remark this answers is not in the document]
Comment (adjackura) [reply agentendpoint/patch_linux.go]: It's parsed at build time:
https://golang.org/pkg/go/build/#hdr-Build_Constraints

File: agentendpoint/agentendpoint.go (hunk 1 above)
Comment (adjackura) [reply agentendpoint/agentendpoint.go]: The only time we want to exit is if startRes.GetTask() == nil as that means there is no further task

**Pull request closed without merging**